### PR TITLE
Pull request workflow documentation.

### LINKED ISSUE #490: Document pull request workflow
Via comment on #489. Basically my feeling is it should go something like this, just need to write some textual explanation:

Do once:
```
git remote add upstream git@github.com:bartvm/blocks.git
```

Verify that `git remote -v |grep origin` points at your own fork.

When beginning a new pull request:
```
git fetch upstream
git checkout -b [ticket_branch_name] upstream/master
git push -u origin ticket_branch_name:ticket_branch_name
```

`git add -p` and `git commit` until satisfied.

```
git push origin ticket_branch_name
```

Send pull request from web interface. **Put 'Fixes #NNN' in the description field.**

----

Also document how to do a rebase.

### PR DESCRIPTION
**Pull request by dwf**
Fixes #490.

### CODE REVIEW

**Comment by bartvm** [comment]
Looks good to me. We might want to add a rebasing instruction later on.

**Comment by dwf** [comment]
Just added basic rebase instructions, punting to GitHub for more details.

Also fixed a Sphinx build error [here](https://github.com/dwf/blocks/commit/73c861ef715a1ed7b570b2e2cd03e27eccbea014#diff-00ab472a96028e9c3bcd77657d6c3701R246).

File: docs/development/pullrequest.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+master. To do this, run (while you have your branch checked out)
+
+.. code-block:: bash
+
+    $ git fetch upstream && git rebase upstream/master
+
+You may encounter an error message about one or more *conflicts*. See
+`GitHub's help page on the subject`_. Note that after a rebase you will
+usually have to overwrite previous commits on your fork's copy of the
+branch with `git push --force`.
```
Comment (bartvm) [inline docs/development/pullrequest.rst]: *``

File: docs/development/pullrequest.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+Working on your pull request
+----------------------------
+
+Make modifications, stage them, and commit them
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Repeat until satisfied:
+
+* Make some modifications to the code
+* Stage them using `git add` (`git add -p` is particularly useful)
```
Comment (bartvm) [inline docs/development/pullrequest.rst]: *``

Comment (dwf) [inline docs/development/pullrequest.rst]: 37f6c8b

**Comment by dwf** [comment]
Squashed a lot of tiny nuisance commits and separated out the build problem fix into a separate commit, and renamed the file to be more consistent with existing stuff.

**Comment by bartvm** [comment]
Okay, merge at will as far as I'm concerned, I have a feeling it's going to be my favourite link in CCW code reviews.

**Comment by dwf** [comment]
It's almost as if this documentation was borne of a desire to save myself typing...

Dr. Evil

**Pull request merged**